osu.Game/Database/OnlineLookupCache.cs

```diff
@@ -127,7 +127,7 @@ private void performLookup()
 
             // rather than queueing, we maintain our own single-threaded request stream.
             // todo: we probably want retry logic here.
-            api.Perform(request);
+            await api.PerformAsync(request).ConfigureAwait(false);
```

Comment:
This change looks... sort of unrelated? Does it have anything to do with the topic of this PR (test request handlers firing after disposal), or was this just bundled with the rest of them?

Reply:
It was required for the alternative version I mentioned, but can be reverted if required. I'd argue it's more correct but potentially not worth changing in this PR?

Reply:
I think it's fine, can stay given it's required. Was just curious